model/task_lifecycle.go

```diff
@@ -31,111 +31,114 @@ type StatusChanges struct {
+				if err != nil {
+					catcher.Wrapf(err, "error getting tasks '%s' depends on", t.Id)
+				}
+				if t.IsPartOfSingleHostTaskGroup() {
+					for _, dep := range deps {
+						// reset any already finished tasks in the same task group
+						if dep.TaskGroup == t.TaskGroup && t.TaskGroup != "" && dep.IsFinished() {
+							if err = resetTask(dep.Id, caller, false); err != nil {
+								catcher.Wrapf(err, "error resetting dependency '%s'", dep.Id)
+							}
+						} else {
+							tasksToActivate = append(tasksToActivate, dep)
 						}
-					} else {
-						tasksToActivate = append(tasksToActivate, dep)
 					}
+				} else {
+					tasksToActivate = append(tasksToActivate, deps...)
+				}
+			}
+
+			// Investigating strange dispatch state as part of EVG-13144
+			if !utility.IsZeroTime(t.DispatchTime) && t.Status == evergreen.TaskUndispatched {
+				if err := resetTask(t.Id, caller, false); err != nil {
+					catcher.Wrap(err, "error resetting task")
 				}
 			} else {
-				tasksToActivate = append(tasksToActivate, deps...)
+				tasksToActivate = append(tasksToActivate, originalTasks...)
 			}
-		}
 
-		// Investigating strange dispatch state as part of EVG-13144
-		if !utility.IsZeroTime(t.DispatchTime) && t.Status == evergreen.TaskUndispatched {
-			if err := resetTask(t.Id, caller, false); err != nil {
-				return errors.Wrap(err, "error resetting task")
+			// If the task was not activated by step back, and either the caller is not evergreen
+			// or the task was originally activated by evergreen, deactivate the task
+		} else if !evergreen.IsSystemActivator(caller) || evergreen.IsSystemActivator(t.ActivatedBy) {
+			var err error
+			// deactivate later tasks in the group as well, since they won't succeed without this one
+			if t.IsPartOfSingleHostTaskGroup() {
+				tasksInGroup, err := task.FindTaskGroupFromBuild(t.BuildId, t.TaskGroup)
+				if err != nil {
+					catcher.Wrapf(err, "error finding task group '%s'", t.TaskGroup)
+				}
+				for _, taskInGroup := range tasksInGroup {
+					if taskInGroup.TaskGroupOrder > t.TaskGroupOrder {
+						originalTasks = append(originalTasks, taskInGroup)
+					}
+				}
+			}
+			if t.Requester == evergreen.MergeTestRequester {
+				if err = DequeueAndRestartForTask(nil, &t, message.GithubStateError, caller, fmt.Sprintf("deactivated by '%s'", caller)); err != nil {
+					catcher.Add(err)
+				}
 			}
-		} else {
 			tasksToActivate = append(tasksToActivate, originalTasks...)
+		} else {
+			return nil
 		}
-		if err := task.ActivateTasks(tasksToActivate, time.Now(), caller); err != nil {
-			return errors.Wrapf(err, "can't activate tasks")
-		}
-
-		if t.DistroId == "" && !t.DisplayOnly {
-			grip.Critical(message.Fields{
-				"message": "task is missing distro id",
-				"task_id": t.Id,
-			})
+		if t.IsPartOfDisplay() {
+			if err := UpdateDisplayTaskForTask(&t); err != nil {
+				catcher.Wrap(err, "problem updating display task")
+			}
 		}
+	}
+	if catcher.HasErrors() {
```

Comment:
We've already handled some resetting and etc above, right? So I'm wondering if we should continue to the lower else/if and status updates, and just return the catcher last.

Reply:
True since we are already resetting and potentially dequeuing / restarting some tasks in this block it would probably be better if we allow all of our status updates run for the tasks that do get collected successfully instead of prematurely returning an err.